File: src/main/java/org/cubexmc/metro/util/MetroConstants.java
package org.cubexmc.metro.util;

import org.bukkit.NamespacedKey;
import org.cubexmc.metro.Metro;

/**
 * Shared constants used across runtime components.
 */
public final class MetroConstants {

    private MetroConstants() {
    }

    public static final String METRO_MINECART_NAME = "MetroMinecart";
    public static final String SCOREBOARD_OBJECTIVE = "metro";

    // PDC key for identifying metro minecarts
    private static NamespacedKey minecartKey;

    public static void initialize(Metro plugin) {
        if (minecartKey == null) {
            minecartKey = new NamespacedKey(plugin, "is_metro");
        }
    }

    public static NamespacedKey getMinecartKey() {
        return minecartKey;
    }
}

File: src/main/java/org/cubexmc/metro/util/MetroConstants.kt
package org.cubexmc.metro.util

import org.bukkit.NamespacedKey
import org.cubexmc.metro.Metro

/**
 * Shared constants used across runtime components.
 */
object MetroConstants {
    const val METRO_MINECART_NAME: String = "MetroMinecart"
    const val SCOREBOARD_OBJECTIVE: String = "metro"

    // PDC key for identifying metro minecarts
    private var minecartKey: NamespacedKey? = null

    // PDC key for identifying items that belong to a Metro GUI and must never
    // end up in a player inventory
    private var guiItemKey: NamespacedKey? = null

    @JvmStatic
    fun initialize(plugin: Metro) {
        if (minecartKey == null) {
            minecartKey = NamespacedKey(plugin, "is_metro")
        }
        if (guiItemKey == null) {
            guiItemKey = NamespacedKey(plugin, "gui_item")
        }
    }

    @JvmStatic
    fun getMinecartKey(): NamespacedKey? = minecartKey

    @JvmStatic
    fun getGuiItemKey(): NamespacedKey? = guiItemKey
}
